src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -272,39 +324,58 @@ def print_help(self):
         )
 
-    def report(self):
-        if self.args.format == 'pretty':
-            self.strategy.report_pretty(self.filtered_devices)
-        elif self.args.format == 'json':
-            self.strategy.report_json(self.filtered_devices)
+    def report(self, plan):
+        if self.args.format == 'json':
+            print(json.dumps([osd.report_json() for osd in plan]))
+        elif self.args.format == 'json-pretty':
+            print(json.dumps([osd.report_json() for osd in plan], indent=4,
+                       sort_keys=True))
         else:
-            raise RuntimeError('report format must be "pretty" or "json"')
-
-    def execute(self):
-        if not self.args.yes:
-            self.strategy.report_pretty(self.filtered_devices)
-            terminal.info('The above OSDs would be created if the operation continues')
-            if not prompt_bool('do you want to proceed? (yes/no)'):
-                devices = ','.join([device.abspath for device in self.args.devices])
-                terminal.error('aborting OSD provisioning for %s' % devices)
-                raise SystemExit(0)
-
-        self.strategy.execute()
-
-    def _get_strategy(self):
-        strategy = get_strategy(self.args, self.args.devices)
-        unused_devices, self.filtered_devices = filter_devices(self.args)
-        if not unused_devices and not self.args.format == 'json':
-            # report nothing changed
-            mlogger.info("All devices are already used by ceph. No OSDs will be created.")
-            raise SystemExit(0)
+            report = ''
+            report += templates.total_osds.format(total_osds=len(plan))
+
+            report += templates.osd_component_titles
+            for osd in plan:
+                report += templates.osd_header
+                report += osd.report()
+
+            print(report)
+
+    def _check_slot_args(self):
+        '''
+        checking if -slots args are consistent with other arguments
+        '''
+        if self.args.data_slots and self.args.osds_per_device:
+            if self.args.data_slots < self.args.osds_per_device:
+                raise ValueError('data_slots is smaller then osds_per_device')
+        # TODO this needs more thought.
+        # for slot in ['block_db_slots', 'block_wal_slots', 'journal_slots']:
+        #     slot_value = getattr(self.args, slot, None)
+        #     if slot_value:
+        #         if slot_value < len(self.args.devices):
+        #             raise ValueError('{} is smaller then osds_per_device')
+
+    def _sort_rotational_disks(self):
+        '''
+        Helper for legacy auto behaviour.
+        Sorts drives into rotating and non-rotating, the latter being used for
+        db or journal.
+        '''
+        mlogger.warning('DEPRECATION NOTICE')
+        mlogger.warning('You are using the legacy automatic disk sorting behavior')
+        mlogger.warning('The Pacific release will change the default to --notauto')
```

Comment:
The docs say that `--no-auto` will go away and only `--auto` will stay. Just a typo maybe?

Reply:
I think it doesn't, but I'll improve the formatting a bit.